fdbclient/ThreadSafeTransaction.actor.cpp

```diff
@@ -260,13 +260,15 @@ ThreadFuture< Void > ThreadSafeTransaction::commit() {
 
 Version ThreadSafeTransaction::getCommittedVersion() {
 	// This should be thread safe when called legally, but it is fragile
-	Version v = tr->getCommittedVersion();
-	return v;
+	return tr->getCommittedVersion();
+}
+
+ThreadFuture<int64_t> ThreadSafeTransaction::getApproximateSize() {
+	return onMainThread([this]() -> Future<int64_t> { return tr->getApproximateSize(); });
 }
 
 ThreadFuture<Standalone<StringRef>> ThreadSafeTransaction::getVersionstamp() {
-	ReadYourWritesTransaction *tr = this->tr;
-	return onMainThread([tr]() -> Future < Standalone<StringRef> > {
+	return onMainThread([this]() -> Future < Standalone<StringRef> > {
```

Comment:
I think the pattern we were using was intentional, though I forget exactly why. I would keep that pattern both here and in your newly added function.

Reply:
Yes. I think the reason is that "this" may be destroyed, while "tr" is still valid. I've reverted the change.